src/main/java/org/sonarlint/intellij/server/SonarLintHttpServer.kt

```diff
@@ -0,0 +1,182 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.server
+
+import com.intellij.notification.NotificationGroup.Companion.balloonGroup
+import io.netty.bootstrap.ServerBootstrap
+import io.netty.buffer.Unpooled
+import io.netty.channel.ChannelFutureListener
+import io.netty.channel.ChannelHandlerContext
+import io.netty.channel.ChannelInitializer
+import io.netty.channel.EventLoopGroup
+import io.netty.channel.SimpleChannelInboundHandler
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.SocketChannel
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.handler.codec.http.DefaultFullHttpResponse
+import io.netty.handler.codec.http.FullHttpResponse
+import io.netty.handler.codec.http.HttpContent
+import io.netty.handler.codec.http.HttpHeaderNames
+import io.netty.handler.codec.http.HttpHeaderValues
+import io.netty.handler.codec.http.HttpMethod
+import io.netty.handler.codec.http.HttpObject
+import io.netty.handler.codec.http.HttpRequest
+import io.netty.handler.codec.http.HttpRequestDecoder
+import io.netty.handler.codec.http.HttpResponseEncoder
+import io.netty.handler.codec.http.HttpResponseStatus
+import io.netty.handler.codec.http.HttpUtil
+import io.netty.handler.codec.http.HttpVersion
+import io.netty.handler.codec.http.LastHttpContent
+import io.netty.handler.logging.LogLevel
+import io.netty.handler.logging.LoggingHandler
+import io.netty.util.CharsetUtil
+import org.sonarlint.intellij.util.SonarLintUtils
+import java.net.BindException
+import kotlin.concurrent.thread
+
+const val STARTING_PORT = 63000
+const val INVALID_REQUEST = "Invalid request."
+const val UNKNOWN_INTELLIJ_FLAVOR = "Unknown IntelliJ flavor."
+const val PORT_RANGE = 3
+const val ENVIRONMENT_ENDPOINT = "/sonarlint/environment"
+
+
+
+object SonarLintHttpServer {
+
+    fun start() {
+        tryToStart(0)
+    }
+
+    fun tryToStart(numberOfAttempts: Int) {
+        try {
+            actualStart(STARTING_PORT + numberOfAttempts)
+        } catch (e: BindException) {
+            if (numberOfAttempts < PORT_RANGE) {
+                tryToStart(numberOfAttempts + 1)
+            } else {
+                throw RuntimeException("Couldn't start SonarLint server in port range: $STARTING_PORT - ${STARTING_PORT + PORT_RANGE}")
```

Comment:
Is this exception caught somewhere ? How would we show the user that it failed ?